components/server/src/test/java/com/hotels/styx/server/netty/connectors/HttpResponseWriterTest.java

```diff
@@ -375,6 +388,40 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpResponse response) th
         assertThat(channelRead.get(), is(true));
     }
 
+    @Test
+    public void logsSentAndAcknowledgedBytes() {
+        EmbeddedChannel ch = new EmbeddedChannel(
+                new SimpleChannelInboundHandler<HttpResponse>() {
+                    @Override
+                    protected void channelRead0(ChannelHandlerContext ctx, HttpResponse response) throws Exception {
+                        HttpResponseWriter writer = new HttpResponseWriter(ctx);
+                        CompletableFuture<Void> future = writer.write(response);
+                        assertThat(future.isDone(), is(false));
+
+                        contentObservable.onNext(copiedBuffer("aaa", UTF_8));
+                        assertThat(future.isDone(), is(false));
+
+                        contentObservable.onNext(copiedBuffer("bbbb", UTF_8));
+                        assertThat(future.isDone(), is(false));
+
+                        contentObservable.onError(new TransportLostException(new InetSocketAddress("localhost", 5050), newOriginBuilder("localhost", 5050).build()));
```

Comment:
As a general rule, we might want to use InetAddress.getLoopbackAddress()  instead of "localhost"  so  we never get an error when trying to resolve "localhost".  In this case the resolution of "localhost"  wouldn't throw an error if it can be resolved but the loopbackadress would still prevent any resolution from happening.

Reply:
👍  Good point.